controllers/ArticleController.js

```diff
@@ -80,32 +80,76 @@ class ArticleController {
   }
 
   /**
-   * get all articles created
+   * get all articles created and use the query
+   * if provided to implement pagination
    * @param {object} req - request object
    * @param {object} res - response object
    * @returns {object} - the found article from database or empty if not found
    */
   static listAllArticles(req, res) {
+    /** get the page number from the query
+     *  and if not provided make it a default
+     *  value  of 1
+     */
+    let { page } = req.query;
+    let limit, offset;
+
+    if (page) {
+      /** convert the page query to an integer
+       *  check that the integer is greater
+       *  than zero and is not a float or string (NaN)
+       */
+      page = Number(page);
+      /** calculate offset and limit */
+      limit = 10;
```

Comment:
I do not feel limit should be hardcoded, it should be extracted from the query parameter. It is okay to initialize it with a number like you have here, but it should not be fixed. Please consider making this change.

Reply:
The idea of pagination is not to display everything at once but rather in batches, you can't give the user the ability to determine the number of articles, that will defeat the purpose of pagination. for every page requested 10 articles will be displayed.